syft/frameworks/torch/mpc/falcon/falcon_helper.py

```diff
@@ -0,0 +1,135 @@
+import torch
+import syft
+
+from syft.frameworks.torch.tensors.interpreters.replicated_shared import ReplicatedSharingTensor
+
+from typing import Union
+
+
+class FalconHelper:
+    XOR_EXPECTED_RING_SIZE = 2
+
+    @classmethod
+    def unfold(cls, image, kernel_size, padding):
+        return cls.__switch_public_private(
+            image, cls.__public_unfold, cls.__private_unfold, kernel_size, padding
+        )
+
+    @staticmethod
+    def __public_unfold(image, kernel_size, padding):
+        image = image.double()
+        image = torch.nn.functional.unfold(image, kernel_size=kernel_size, padding=padding)
+        image = image.long()
+        return image
+
+    @staticmethod
+    def __private_unfold(image, kernel_size, padding):
+        return image.unfold(kernel_size, padding)
+
+    @staticmethod
+    def xor(
+        value: Union[int, ReplicatedSharingTensor], other: Union[int, ReplicatedSharingTensor]
+    ) -> ReplicatedSharingTensor:
+        """
+        Compute the XOR value between value and other.
+        If value and other are both ints we should use the "^" operator.
+
+        Args:
+            value (ReplicatedSharingTensor or Int): RST with ring size of 2 or Int value in {0, 1}
```

Comment:
value should be always RST, other could be RST for private xor, or long tensor or int for public xor

Reply:
Hm, I made it like this to be able to have ```xor(int, RST)``` - I can remove it